src/main/scala/net/bdew/ae2stuff/machines/wireless/TileWireless.scala

```diff
@@ -123,23 +150,55 @@ class TileWireless
     false
   }
 
+  def setHubPowerUse(power: Double): Unit = {
+    hubPowerUsage += power
+    this.setIdlePowerUse(hubPowerUsage)
+  }
+
+  def getHubChannels: Int = {
+    var channels = 0
+    connectionsList foreach { that =>
+      channels += that.connection.getUsedChannels
+    }
+    channels
+  }
+
   def breakConnection(): Unit = {
     if (connection != null)
       connection.destroy()
     connection = null
-    setIdlePowerUse(0d)
     getLink foreach { other =>
-      other.connection = null
-      other.setIdlePowerUse(0d)
-      if (worldObj.blockExists(other.xCoord, other.yCoord, other.zCoord))
-        worldObj.setBlockMetadataWithNotify(
-          other.xCoord,
-          other.yCoord,
-          other.zCoord,
-          0,
-          3
+      if (other.isHub) {
+        other.connectionsList = other.connectionsList.filterNot(_ == this)
+        other.setHubPowerUse(-getIdlePowerUsage)
+        if (
+          other.connectionsList.isEmpty && worldObj.blockExists(
+            other.xCoord,
+            other.yCoord,
+            other.zCoord
+          )
         )
+          worldObj.setBlockMetadataWithNotify(
+            other.xCoord,
+            other.yCoord,
+            other.zCoord,
+            0,
+            3
+          )
+      } else {
+        other.connection = null
+        other.setIdlePowerUse(0d)
+        if (worldObj.blockExists(other.xCoord, other.yCoord, other.zCoord))
+          worldObj.setBlockMetadataWithNotify(
+            other.xCoord,
+            other.yCoord,
+            other.zCoord,
+            0,
+            3
+          )
+      }
     }
+    setIdlePowerUse(0d)
```

Comment:
Does this need to check for hub? Right now if I'm reading this correctly it will set the hub power to zero on only one connection being broken

Reply:
Hub never call breakConnection, because actually not connected to anything, other connectors connected to him. When connector connecting/disconnecting it do all work and then just tell hub add it to list or remove. When you break hub, it tell to all in list break connection to it. That why there no hub to hub connection xD